pkg/cluster/cache/redis.go

```diff
@@ -0,0 +1,100 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha2"
+	"github.com/goharbor/harbor-operator/pkg/k8s"
+	"github.com/goharbor/harbor-operator/pkg/lcm"
+	redisOp "github.com/spotahome/redis-operator/api/redisfailover/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+var (
+	redisFailoversGVR = redisOp.SchemeGroupVersion.WithResource(redisOp.RFNamePlural)
+)
+
+// NewRedisController is constructor for redis controller.
+func NewRedisController() lcm.Controller {
+	return &RedisController{}
+}
+
+// RedisController implements lcm.Controller interface.
+type RedisController struct {
+	Ctx                context.Context
+	Client             k8s.Client
+	DClient            k8s.DClient
+	Recorder           record.EventRecorder
+	Log                logr.Logger
+	Scheme             *runtime.Scheme
+	RedisConnect       *RedisConnect
+	ResourceManager    ResourceManager
+	expectCR, actualCR *redisOp.RedisFailover
+}
+
+// Apply creates/updates/scales the resources, like kubernetes apply operation.
+func (rc *RedisController) Apply(cluster *v1alpha2.HarborCluster) (*lcm.CRStatus, error) {
+	rc.Client.WithContext(rc.Ctx)
+	rc.DClient.WithContext(rc.Ctx)
+
+	crdClient := rc.DClient.WithResource(redisFailoversGVR).WithNamespace(cluster.Namespace)
```

Comment:
There need to check whether `cluster.inClusterCache` defined. If not defined. check the external Redis health and return the `CRStatus`